kong/init.lua

```diff
@@ -881,19 +881,21 @@ function Kong.init_worker()
   end
 
   if kong.clustering then
-    -- full sync dp
-
-    local is_dp_full_sync_agent = process.type() == "privileged agent" and not kong.sync
-
-    if is_control_plane(kong.configuration) or -- CP needs to support both full and incremental sync
-      is_dp_full_sync_agent -- full sync is only enabled for DP if incremental sync is disabled
-    then
+    if is_control_plane(kong.configuration) then-- CP needs to support both full and incremental sync
       kong.clustering:init_worker()
-    end
+    
+    -- full sync is only enabled for DP if incremental sync is disabled
+    elseif is_data_plane(kong.configuration) and not kong.sync then
+      local using_dedicated = kong.configuration.dedicated_config_processing
+      if using_dedicated and process.type() == "privileged agent" then
+        -- full sync dp agent
+        kong.clustering:init_worker()
+        return
 
-    -- DP full sync agent skips the rest of the init_worker
-    if is_dp_full_sync_agent then
-      return
+      elseif not using_dedicated then
```

Comment:
```suggestion
      end
      
      if not using_dedicated then
```

Reply:
These `if-else` is too complicated to understand, could we simplify it or add some comments to explain it?

Reply:
@chronolaw I've already named each branch and commented on them. Any concrete suggestion is welcomed.

Reply:
I think this structure and the comments are already quite clear. However, I feel that maybe modify this part 

```
 if using_dedicated and process.type() == "privileged agent" then
        -- full sync dp agent
        kong.clustering:init_worker()
        return

      elseif not using_dedicated then         <<------ could be ended here
```

to the following 

```
 if using_dedicated and process.type() == "privileged agent" then
        -- full sync dp agent
        kong.clustering:init_worker()
        return
 end

 if not using_dedicated then
```